lib/glific/triggers.ex

```diff
@@ -155,16 +155,63 @@ defmodule Glific.Triggers do
   """
   @spec create_trigger(map()) :: {:ok, Trigger.t()} | {:error, Ecto.Changeset.t()}
   def create_trigger(attrs) do
-    with {:ok, trigger} <-
+    with {:ok, valid_attrs} <- validate_new_trigger(attrs),
```

Comment:
I don't get it. Why we added the validation code here instead of Trigger Structs?

Reply:
@pankaj-ag Thanks for letting me know. I have made requested changes